### ISSUE TITLE
Problems with reverse proxy after upgrading to 7.12.3

### ISSUE THREAD

**Issue by rickardliljeberg** [umbraco/Umbraco-CMS#3418]
### Previously working solution

I have a project set up with a reverse proxy in front of it and this used to work fine. I was running 7.8 until I recently upgraded to 7.12.3

The solution I had in place looked like this (owin middleware)
As you can see in the watch at the bottom it has correctly gotten the x-forwarded-host from the reverse proxy. And this used to work fine in 7.7.8
Note: ec-web.se is not a real domain, it's set up in my hosts file as a test domain
![owin_startup](https://user-images.githubusercontent.com/3726818/47417389-2b75fc00-d778-11e8-8707-227ba1facb0f.png)

### New problem in 7.12.3

However if I now go to a controller (that overrides RenderMvcController) and there look at "Request.Url" it reports:

> http://ec-web.se:8788/

So it replaced the host but not the port. And no matter how I try I cannot make it replace/remove the port.
This incorrect Url makes some of my code fail.

### Not finding correct root node

This project have three different root nodes with different urls and now it no longer picks the correct root node (it simply picks the first).

To test fully I have added all these three hostnames to the node I want
![image](https://user-images.githubusercontent.com/3726818/47417793-151c7000-d779-11e8-90a7-231e61707820.png)

if I browse straight to **localhost:8788** (thus bypassing the reverse proxy and going straight to the project) it works and displays the correct root node.
However if I browse to **ec-web.se** (which in turn redirects all traffic to 127.0.0.1:8788 as a reverse proxy) it does not work.

However if I now also add the following
![image](https://user-images.githubusercontent.com/3726818/47418014-8b20d700-d779-11e8-8487-816587412420.png)

I can browse to ec-web.se and it works.

## TL;DR

**Issue 1** In 7.12.3 I cannot get ridd of the port in owin middlewhere so code in controllers using **Request.Url** fails due to url containing port.

**Issue 2** Umbraco does not respect x-forwarded-host when selecting root node.

**Comment by nul800sebastiaan** [comment]
I'm sorry to hear that upgrading has caused this problem. I've asked Shannon and a quick reply was: this might be a change made in Owin, not exactly in Umbraco.

As for your second issue, this I believe is covered in depth over here: https://issues.umbraco.org/issue/U4-9229

If you've found a workaround by now it would be great if you could post it up. Unfortunately for you, we've not seen this issue before and it will take some effort to resolve it. If you have fix we could put in Umbraco then that would be awesome, currently it doesn't fit in our schedule to look at this properly.

I've marked this as "Up for grabs" so that you or someone else coming along could create a pull request for it.

**Comment by nul800sebastiaan** [comment]
This issue has been in the `up for grabs` state for quite a while now and it seems nobody is ready to pick this one up.

For now we'll close this issue to prevent the list of `up for grabs` from becoming very stale. We're happy to re-open it if someone still thinks it's good to work on this. 

If anyone is about to pick this issue up to fix it, make sure to test first if you can reproduce the problem in the latest version before you start to work on it.

Thanks!
Sebastiaan on behalf of Umbraco HQ.
